src/stream/src/executor/eowc/eowc_gap_fill.rs

```diff
@@ -71,20 +71,23 @@ struct ExecutionVars<S: StateStore> {
 }
 
 impl<S: StateStore> ExecutorInner<S> {
-    fn generate_filled_rows(
-        prev_row: &OwnedRow,
-        curr_row: &OwnedRow,
+    #[try_stream(ok = StreamChunk, error = ExprError)]
+    async fn generate_filled_rows<'a>(
+        prev_row: &'a OwnedRow,
+        curr_row: &'a OwnedRow,
         time_column_index: usize,
-        fill_columns: &HashMap<usize, FillStrategy>,
+        fill_columns: &'a HashMap<usize, FillStrategy>,
         interval: risingwave_common::types::Interval,
-        metrics: &GapFillMetrics,
-    ) -> Result<Vec<OwnedRow>, ExprError> {
-        let mut filled_rows = Vec::new();
-        let (Some(prev_time_scalar), Some(curr_time_scalar)) = (
-            prev_row.datum_at(time_column_index),
-            curr_row.datum_at(time_column_index),
-        ) else {
-            return Ok(filled_rows);
+        metrics: &'a GapFillMetrics,
+        chunk_builder: &'a mut StreamChunkBuilder,
+    ) {
+        let Some((prev_time_scalar, curr_time_scalar)) = (|| {
+            Some((
+                prev_row.datum_at(time_column_index)?,
+                curr_row.datum_at(time_column_index)?,
+            ))
+        })() else {
+            return Ok(());
```

Comment:
We should not return `Ok(())` if the row schema is mismatched. It should return an error.